lib/graphql_devise/concerns/controller_methods.rb

```diff
@@ -90,7 +90,14 @@ def redirect_headers(token_info, redirect_header_options)
       end
 
       def find_resource(field, value)
-        if resource_class.try(:connection_config).try(:[], :adapter).try(:include?, 'mysql')
+        connection_config_method =
```

Comment:
I think you will need a different conditional for this as the new method does not return a hash directly. This won't make the tests fail (we need specs for mysql @00dav00 ) because it uses try a lot 😬 . But if actually running mysql I won't have the desired behavior. Please change it to something like this:

```ruby
def mysql_db?
  if Gem::Version.new(ActiveRecord.version) < Gem::Version.new('6.1.0.rc1')
    resource_class.try(:connection_config).try(:[], :adapter).try(:include?, 'mysql')
  else
    resource_class.connection_db_config.configuration_hash[:adapter].try(:include?, 'mysql')
  end
end
```

Then the next conditional can just be
```ruby
if mysql_db?
# fix for mysql default case insensitivity
  resource_class.where("BINARY #{field} = ? AND provider= ?", value, provider).first
  ......

Reply:
hi @mcelicalderon,
thanks for your review. I thought it's not breaking change in rails and didn't notice change in their code 🙈 

I wonder if we can use `resource_class.connection.adapter_name == 'Mysql2'` instead of this complex connection config diggin?

https://github.com/rails/rails/blob/master/activerecord/lib/active_record/connection_adapters/mysql2_adapter.rb#L34

In my local env it returns this for pg:
```
irb(main):012:0> User.connection.adapter_name
=> "PostgreSQL"
```

Reply:
You are right, that is actually simpler and the the method definition hasn't changed in 15 years 😆 [here](https://github.com/rails/rails/blame/3397924e6697631970bb50c6475d2922656311aa/activerecord/lib/active_record/connection_adapters/abstract_adapter.rb#L262). So yes, please do that. We copied this from DTA and didn't ask questions. Please downcase and check for mysql. No need for an additional method

```ruby
if resource_class.connection.adapter_name.downcase.includes?('mysql')